job/steps/semantic_similarity.py

```diff
@@ -46,3 +53,13 @@ def preprocess_data(site: Site, article_data: List[Dict[str]], interactions_data
     df = df_metadata.merge(df_data, how="inner", on="external_id")
 
     return df
+
+
+def generate_embeddings(train_data: pd.DataFrame) -> np.ndarray:
+    """
+    Given article data DataFrame with a text column, create article-level text embeddings.
+    """
+    texts = train_data["text"].tolist()
+    model = SentenceTransformer(PRETRAINED_MODEL_NAME, device="cpu")
```

Comment:
❓ Will this always be `cpu` or does it need to be configurable?

Reply:
We don't need to go beyond CPU for now. We already don't need GPU for Spotlight to create embeddings, and SentenceTransformer takes a lot less time to do so.